estate/models/estate_property.py

```diff
@@ -0,0 +1,108 @@
+            ('offer_accepted', "Offer Accepted"),
+            ('sold', "Sold"),
+            ('cancelled', "Cancelled")
+        ]
+    )
+    property_type_id = fields.Many2one('estate.property.types', string="Property Type")
+    salesman_id = fields.Many2one('res.users', string="Salesman", default=lambda self: self.env.user)
+    buyer_id = fields.Many2one('res.partner', string="Buyer")
+    tag_ids = fields.Many2many('estate.property.tag', string="Property Tags")
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string="Offers")
+    total_area = fields.Float(compute="_compute_total_area", string="Total Area")
+    best_price = fields.Float(compute="_compute_best_price", string="Best Price")
+
+    _sql_constraints = [
+        ('check_expected_price', 'CHECK(expected_price > 0)',
+         "The expected price should be greater than 0."),
+        ('check_selling_price', 'CHECK(buyer_id IS NULL OR selling_price > 0)',
+         "The selling price should be greater than 0."),
+    ]
+
+    @api.constrains('selling_price', 'expected_price')
+    def _check_prices_difference(self):
+        for record in self:
+            if record.buyer_id and float_compare(record.selling_price, 0.9 * record.expected_price, 5) == -1:
+                raise UserError(_("Selling price cannot be lower than 90% of the expected price!"))
+
+    @api.depends('garden_area', 'living_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.garden_area + record.living_area
+
+    @api.depends('offer_ids.price')
+    def _compute_best_price(self):
+        for record in self:
+            prices = record.offer_ids.mapped('price')
+
+            record.best_price = max(prices) if len(prices) > 0 else 0
+
+    @api.onchange('garden')
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = ''
+
+    def action_cancel(self):
+        for record in self:
+            if record.state == 'cancelled':
+                raise UserError(_("Property already cancelled!"))
+            elif record.state == 'sold':
+                raise UserError(_("A sold property cannot be cancelled!"))
+            else:
+                self.state = 'cancelled'
+
+        return True
+
+    def action_sold(self):
+        for record in self:
+            if record.state == 'sold':
+                raise UserError(_("Property already sold!"))
+            elif record.state == 'cancelled':
+                raise UserError(_("A cancelled property cannot be sold!"))
+            else:
+                self.state = 'sold'
+
+        return True
+
+    @api.ondelete(at_uninstall=False)
+    def prevent_deletion(self):
+        for record in self:
+            if record.state != 'new' and record.state != 'cancelled':
```

Comment:
Easier to add and remove values in the future
```suggestion
            if record.state not in ['new', 'cancelled']:
```